module/os/freebsd/spl/spl_kstat.c

```diff
@@ -304,7 +304,7 @@ __kstat_create(const char *module, int instance, const char *name,
 	} else {
 		ksp->ks_data = kmem_zalloc(ksp->ks_data_size, KM_SLEEP);
 		if (ksp->ks_data == NULL) {
-			kmem_free(ksp, sizeof (*ksp));
+			free(ksp, M_KSTAT);
```

Comment:
While the free() is obviously incorrect, I don't think this code is ever reachable, considering KM_SLEEP.  And after that setting ksp = NULL should probably crash on line 328.  So I'd probably remove this error check completely to fix both issues.